heron/packing/src/java/com/twitter/heron/packing/binpacking/FirstFitDecreasingPacking.java

```diff
@@ -151,6 +155,36 @@ public PackingPlan pack() {
     return new PackingPlan(topology.getId(), containerPlans);
   }
 
+  /**
+   * Get a new packing plan given an existing packing plan and component-level changes
+   *
+   * @return new packing plan
+   */
+  public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> componentChanges) {
+    // Get the instances using FFD allocation
+    Map<Integer, List<String>> ffdAllocation =
+        getFFDAllocation(currentPackingPlan, componentChanges);
+
+    // Construct the PackingPlan
+    Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
+
+    Set<PackingPlan.ContainerPlan> containerPlans = PackingUtils.buildContainerPlans(
+        ffdAllocation, ramMap, defaultInstanceResources, paddingPercentage);
+
+    //merge the two plans
```

Comment:
We should add a comment to clarify that this merge approach only works because the `ffdAllocation` generated above is done by appending new instances into new containers so there is no container intersection in the merge.